util.py

```diff
@@ -469,20 +470,27 @@ def get_instance_id(agentConfig):
 
 
 class Watchdog(object):
-    """Simple signal-based watchdog that will scuttle the current process
-    if it has not been reset every N seconds, or if the processes exceeds
-    a specified memory threshold.
+    """
+    Simple signal-based watchdog. Restart the process when:
+    * no reset was made for more than a specified duration
+    * (optional) a specified memory threshold is exceeded
+    * (optional) a suspicious high activity is detected, i.e. too many resets for a given timeframe.
+
+    **Warning**: Not thread-safe.
     Can only be invoked once per process, so don't use with multiple threads.
     If you instantiate more than one, you're also asking for trouble.
     """
-    def __init__(self, duration, max_mem_mb = None):
+    # Activity history timeframe
+    _RESTART_TIMEFRAME = 60
+
+    def __init__(self, duration, max_mem_mb=None, max_resets=None):
```

Comment:
I don't agree: `max_resets` as a value per second is not a sufficient information to understand how the watchdog behaves. You'd need to know how this rate is computed, i.e. a timeframe length.

Considering this, having  `max_reset` as  a plain threshold versus a rate over a timeframe, is, in my opinion, easier to comprehend.